cli/command.go

```diff
@@ -274,19 +278,42 @@ func (c *BaseCommand) Hidden() bool {
 
 // Prompt prompts the user for a value. If stdin is a tty, it prompts. Otherwise
 // it reads from the reader.
-func (c *BaseCommand) Prompt(msg string) (string, error) {
-	scanner := bufio.NewScanner(io.LimitReader(c.Stdin(), 64*1_000))
+func (c *BaseCommand) Prompt(ctx context.Context, msg string, defaultValue *string) (string, error) {
+	// guarantee Prompt is cancelable with SIGTERM and SIGINT
+	ctx, done := signal.NotifyContext(ctx, syscall.SIGTERM, syscall.SIGINT)
+	defer done()
 
-	if c.Stdin() == os.Stdin && isatty.IsTerminal(os.Stdin.Fd()) {
-		fmt.Fprint(c.Stdout(), msg)
-	}
+	resultChan := make(chan string)
+	errorChan := make(chan error)
+
+	go func() {
+		scanner := bufio.NewScanner(io.LimitReader(c.Stdin(), 64*1_000))
+
+		if c.Stdin() == os.Stdin && isatty.IsTerminal(os.Stdin.Fd()) {
+			fmt.Fprint(c.Stdout(), msg)
+		}
+
+		scanner.Scan()
+
+		if err := scanner.Err(); err != nil {
+			errorChan <- fmt.Errorf("failed to read stdin: %w", err)
+		}
+
+		if defaultValue != nil && strings.TrimSpace(scanner.Text()) == "" {
```

Comment:
Because `scanner.Text()` will return "" for no input and it seemed silly to set "" = "" if no default value is given. I suppose it doesnt matter really tho.

Reply:
Should we just remove defaultValue and let the user do it on their end?

Reply:
Removing default value